src/TorchVision/Utils.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using static TorchSharp.torch;
+using static TorchSharp.torchvision.io;
+
+namespace TorchSharp
+{
+    public static partial class torchvision
+    {
+        public static class utils
+        {
+            public static Tensor make_grid(
+                IEnumerable<Tensor> tensors,
+                long nrow = 8,
+                int padding = 2,
+                bool normalize = false,
+                (double low, double high)? value_range = null,
+                bool scale_each = false,
+                double pad_value = 0.0f)
+            {
+                return make_grid(torch.stack(tensors, dim: 0), nrow, padding, normalize, value_range, scale_each, pad_value);
+            }
+
+            public static Tensor make_grid(
+                Tensor tensor,
+                long nrow = 8,
+                int padding = 2,
+                bool normalize = false,
+                (double low, double high)? value_range = null,
+                bool scale_each = false,
+                double pad_value = 0.0f)
+            {
+                if (tensor.Dimensions == 2) // Single image H x W
+                {
+                    tensor = tensor.unsqueeze(0);
```

Comment:
put the old tensor in a temp, do the unsqueeze, and then dispose the old tensor:

```C#
using var t = tensor;
tensor = tensor.unsqueeze(0);
```

Reply:
Unless, of course, the old tensor is being used outside the function. In that case, create an alias of the input tensor at the beginning, use that throughout, and then have a dispose scope guarding the whole thing. That's probably the easiest way. Make sure to have the call to `.alias()` after you create the dispose scope.

Then, just make sure to call `MoveToOuterDisposeScope()` in the return statements.

Reply:
It's safe to ignore my advice, too, since I would expect most calls to be within a dynamic dispose scope, anyway. However, placing on inside algorithmic functions like this has the advantage of giving back native memory eagerly, which can help at the margins.

Reply:
This is why I said 
> The code might need some additional TorchSharp specific changes.

Reply:
@GilesBathgate 
Thank you and looking forwards to your TorchSharp specific change recommendations